src/Symfony/Bridge/Monolog/Processor/DebugProcessor.php

```diff
@@ -45,14 +63,50 @@ public function __invoke(array $record)
      */
     public function getLogs()
     {
-        return $this->records;
+        if (0 === count($this->recordsRequest)) {
+            return array();
+        }
+
+        if (1 === func_num_args()) {
+            $requestHash = func_get_arg(0);
+            if (isset($this->recordsRequest[$requestHash]['records']) && $requestHash !== $this->masterRequestHash) {
+                return $this->recordsRequest[$requestHash]['records'];
+            }
+        }
+        $records = array_column($this->recordsRequest, 'records');
+
+        return call_user_func_array('array_merge', $records);
     }
 
     /**
      * {@inheritdoc}
      */
     public function countErrors()
     {
-        return $this->errorCount;
+        if (0 === count($this->recordsRequest)) {
+            return 0;
+        }
+
+        if (1 === func_num_args()) {
+            $requestHash = func_get_arg(0);
+            if (isset($this->recordsRequest[$requestHash]['errorCount']) && $requestHash !== $this->masterRequestHash) {
+                return $this->recordsRequest[$requestHash]['errorCount'];
+            }
+        }
+        $errors = array_column($this->recordsRequest, 'errorCount');
+
+        return array_sum($errors);
+    }
+
+    /**
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (null !== $this->masterRequestHash || !$event->isMasterRequest()) {
+            return;
+        }
+
+        $this->masterRequestHash = spl_object_hash($event->getRequest());
```

Comment:
might store the current request hash as well (to use if not passed on the calling side)

Reply:
is passing it on the calling side really needed?

Reply:
I thinking the better move it on `LoggerDataCollector` side.